aten/src/ATen/native/cuda/VolumetricChannelwiseConvolution.cu

```diff
@@ -0,0 +1,859 @@
+  // so its responsible for writing it to gradWeight
+  if (threadIdx.x == 0) {
+    int weightOffset = kW + kernelWidth * (kH + kernelHeight
+      * (kT + kernelTime * ch));
+    gradWeight[weightOffset] = ScalarConvert<accT, T>::to(tval);
+  }
+}
+
+
+static void conv_depthwise3d_cuda_template(
+  Tensor& output,
+  const Tensor& input_,
+  const Tensor& weight,
+  const Tensor& bias,
+  IntArrayRef kernel_size,
+  IntArrayRef stride_size,
+  IntArrayRef pad_size,
+  IntArrayRef dilation_size) {
+  
+    TensorArg input_arg{input_, "input_", 1}, output_arg{output, "output", 2};
+
+    checkAllSameGPU(
+      "conv_depthwise3d_cuda_template", {input_arg, output_arg});
+
+    
+    for (int64_t i = 0; i < input_.ndimension(); i++) {
+      TORCH_CHECK(
+          input_.size(i) > 0,
+          "conv_depthwise3d_cuda(): expected input to have non-empty spatial dimensions, "
+          "but input has sizes ", input_.sizes(),
+          " with dimension ", i, " being empty");
+    }
+    for (int64_t i = 0; i < weight.ndimension(); i++) {
+      TORCH_CHECK(
+          weight.size(i) > 0,
+          "conv_depthwise3d_cuda(): expected weight to have non-empty dimensions, "
+          "but weight has sizes ", weight.sizes(),
+          " with dimension ", i, " being empty");
+    }
+    
+    TORCH_CHECK(
+        (input_.ndimension() == 5),
+        "non-empty 5D tensor expected for input "
+        "but input has size ", input_.ndimension());
+      
+    // Weight Tensor is shape (output_channels, 1, kH, kW)
+    TORCH_CHECK(weight.size(1) == 1);
+
+    // define kernel/stride/pad/dilation
+    int64_t kW = kernel_size[2];
+    int64_t kH = kernel_size[1];
+    int64_t kT = kernel_size[0];
+
+    int64_t dW = stride_size[2];
+    int64_t dH = stride_size[1];
+    int64_t dT = stride_size[0];
+
+    int64_t padW = pad_size[2];
+    int64_t padH = pad_size[1];
+    int64_t padT = pad_size[0];
+
+    int64_t dilationW = dilation_size[2];
+    int64_t dilationH = dilation_size[1];
+    int64_t dilationT = dilation_size[0];
+    
+    int64_t batch_size = input_.size(0);
+    int64_t n_input_plane = input_.size(1);
+    int64_t input_temp = input_.size(2);
+    int64_t input_height = input_.size(3);
+    int64_t input_width = input_.size(4);
+
+    int batchSize = input_.size(0);
+    int length = input_.size(2);
+    int height = input_.size(3);
+    int width = input_.size(4);
+
+    int outputTime = (length + 2 * padT - (dilationT * (kT - 1) + 1)) / dT + 1;
+    int outputHeight = (height + 2 * padH - (dilationH * (kH - 1) + 1)) / dH + 1;
+    int outputWidth = (width + 2 * padW - (dilationW * (kW - 1) + 1)) / dW + 1;
+    int outputChannels = weight.size(0);
+  
+    // Pass empty contiguous tensor and resize it here
+    output.resize_({batchSize, outputChannels, outputTime, outputHeight, outputWidth});
+
+    // make sure input is contiguous
+    const Tensor& input = input_.contiguous();
```

Comment:
The use of const reference lifetime extension here is gratuitous, prefer not to do it